public/app/core/components/FolderFilter/FolderFilter.tsx

```
@@ -66,22 +67,52 @@ async function getFoldersAsOptions(
   setLoading(true);
 
-  const params = {
-    query: searchString,
-    type: DashboardSearchItemType.DashFolder,
-    permission: PermissionLevelString.View,
-  };
+  try {
+    if (config.featureToggles.unifiedStorageSearchUI) {
+      const searcher = getGrafanaSearcher();
+      const queryResponse = await searcher.search({
+        query: searchString,
+        kind: ['folder'],
+        limit: 100,
+        permission: PermissionLevelString.View,
+      });
 
-  // FIXME: stop using id from search and use UID instead
-  const searchHits = await getBackendSrv().search(params);
-  const options = searchHits.map((d) => ({ label: d.title, value: { uid: d.uid, title: d.title } }));
-  if (!searchString || 'dashboards'.includes(searchString.toLowerCase())) {
-    options.unshift({ label: 'Dashboards', value: { uid: 'general', title: 'Dashboards' } });
-  }
+      const options = queryResponse.view.map((item) => ({
+        label: item.name,
+        value: { uid: item.uid, title: item.name }
+      }));
+
+      if (!searchString || 'dashboards'.includes(searchString.toLowerCase())) {
+        options.unshift({ label: 'Dashboards', value: { uid: 'general', title: 'Dashboards' } });
+      }
 
-  setLoading(false);
+      return options;
+    } else {
+      // Use existing backend service search
+      const params = {
+        query: searchString,
+        type: 'folder',
+        permission: PermissionLevelString.View,
+      };
 
-  return options;
+      const searchHits = await getBackendSrv().search(params);
+      const options = searchHits.map((d) => ({ 
+        label: d.title, 
+        value: { uid: d.uid, title: d.title } 
+      }));
+
+      if (!searchString || 'dashboards'.includes(searchString.toLowerCase())) {
+        options.unshift({ label: 'Dashboards', value: { uid: 'general', title: 'Dashboards' } });
+      }
+
+      return options;
+    }
+  } catch (error) {
+    console.error('Failed to fetch folders:', error);
+    return [];
```

maybe throw the error so there is some indication an error occurred in the ui